lisa/notifiers/subtest_aware_notifier.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import IO, Any, Dict, List, Optional, Type, cast
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.messages import (
+    MessageBase,
+    SubTestMessage,
+    TestResultMessage,
+    TestResultMessageBase,
+    TestRunMessage,
+    TestRunStatus,
+    TestStatus,
+)
+from lisa.notifier import Notifier
+from lisa.util import LisaException, constants
+
+# TODO IF these Info classes are not used by any other file, then add leading underscore to their names
+# TODO Check which fields are not required
+class TestCaseRuntimeInfo:
+    def __init__(self) -> None:
+        self.suite_full_name: str = ""
+        self.name: str = ""
+        self.active_subtest_name: Optional[str] = None
+        self.last_seen_timestamp: float = 0.0
+        self.subtest_total_elapsed: float = 0.0
+
+class SubtestAwareNotifier(Notifier):
+    @classmethod
+    def type_name(cls) -> str:
+        # no type_name, not able to import from yaml book.
+        return ""
+
+    def __init__(self, runbook: schema.TypedSchema) -> None:
+        super().__init__(runbook=runbook)
+
+        self._testcases_info: Dict[str, TestCaseRuntimeInfo] = {}
```

Comment:
nit: `__testcases_info`.

Double underscore invokes Python's [name mangling](https://docs.python.org/3/tutorial/classes.html#private-variables) feature, which will avoid child classes from accidentally using the variable.